[DOC] Add CLI section to user documentation. (#12802)
Scille/parsec-cloud: docs/cli/first_steps.rst

.. Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS

.. _doc_cli_first_steps:

===================================

> First steps with Parsec CLI (Linux)

Reviewer: IMO no need to mention Linux here (already mentioned in Install section)
```suggestion
First steps
```

===========

Now that you have installed the Parsec CLI, let's start using it.

> The outputs given in this guide are given as examples. Your output may differ.

Reviewer: Maybe is better to make it an admonition

```suggestion
.. note::

  The outputs displayed in this documentation are examples and 
  may vary from yours depending on your CLI version.
```

Help

> ----

Reviewer: Use "=" for first-level sections.

```suggestion
Help
====
```

This convention is not respected everywhere in our codebase, but is what's mostly used in the hosting docs:

```
=====
Title
=====

H1 section
==========

H2 section
----------

H3 section
^^^^^^^^^^
```

If possible, better avoid more than 3 heading levels.

When in need for information about the Parsec CLI, you can always use `help` as a command
of its own,

> .. code-block:: shell

Reviewer 2: **todo:** Put a disclaimer to indicate that the message bellow is only an example

The content of the help message can and will change easily, so either:

- We remove that block
- We find a way to generate it automatically
- Or we indicate that the block is for information purpose only and is likely to be outdated

That apply for every code-block below

Author: I put that warning in the introduction

Reviewer: IMO putting it in this "First Step" section is enough (see note above). There are no outputs in introduction.rst.

>     ./parsec-cli 3.8.2-a.0+dev help

Reviewer: - Missing `--` before `help`
- No need for version
- Also, it will be nice to avoid `./` if we adapt install method (see https://github.com/Scille/parsec-cloud/pull/12802/changes#r3261017132)

```suggestion
    parsec-cli --help
```

Author: Help is both an option and a command, so the `--` is not needed

    Parsec cli

>     Usage: parsec-cli-3.9.0-rc.1-x86_64-unknown-linux-musl <COMMAND>

Reviewer: ```suggestion
    Usage: parsec-cli <COMMAND>
```

Author: I'd rather keep that as is, as here I want to show an example of the output.

Reviewer: version is still displayed in the output if the binary is named `parsec-cli`?

Author: It isn't I'll fix that

    Commands:
      server              Contains subcommands related to server operations
      device              Contains subcommands related to devices
      invite              Contains subcommands related to invitation
      organization        Contains subcommands related to organization
      user                Contains subcommands related to user
      workspace           Contains subcommands related to workspace
      certificate         Contains subcommands related to certificate
      ls                  List files in a workspace
      rm                  Remove a file from a workspace
      tos                 Contains subcommands related to Term of Service (TOS)
      shared-recovery     Contains subcommands related to shared recovery devices (shamir)
      mount-realm-export  Mount a realm export as a workspace
      help                Print this message or the help of the given subcommand(s)

    Options:
      -h, --help     Print help
      -V, --version  Print version

> or as a options available for any command,

Reviewer: ```suggestion
or as an option available for any command,
```

.. code-block:: shell

>     ./parsec-cli 3.8.2-a.0+dev ls --help

Reviewer: ```suggestion
    parsec-cli ls --help
```

    List files in a workspace

>     Usage: parsec-cli-3.9.0-rc.1-x86_64-unknown-linux-musl ls [OPTIONS] --workspace <WORKSPACE> [PATH]

Reviewer: ```suggestion
    Usage: parsec-cli ls [OPTIONS] --workspace <WORKSPACE> [PATH]
```

    Arguments:
    [PATH]  Path to list [default: /]

    Options:
    -w, --workspace <WORKSPACE>    Workspace ID [env: PARSEC_WORKSPACE_ID=]
        --password-stdin           Read the password from stdin instead of TTY Note: this flag need to be explicitly set, that why it does not have a env var
    -d, --device <DEVICE>          Device ID [env: PARSEC_DEVICE_ID=]
    -c, --config-dir <CONFIG_DIR>  Parsec config directory [env: PARSEC_CONFIG_DIR=] [default: /home/aurelia/.config/parsec3/libparsec]
    -h, --help                     Print help

> group of commands,

Reviewer: ```suggestion
or a group of commands,
```

.. code-block:: shell

>     ./parsec-cli 3.8.2-a.0+dev device --help

Reviewer: ```suggestion
    parsec-cli device --help
```

    Contains subcommands related to devices

>     Usage: parsec-cli-3.9.0-rc.1-x86_64-unknown-linux-musl device <COMMAND>

Reviewer: ```suggestion
    Usage: parsec-cli device <COMMAND>
```

    Commands:
    forget-local            Forget a local device It will still exist on the server but not locally anymore
    list                    List all devices
    change-authentication   Change authentication medium for a device
    export-recovery-device  Export recovery device
    import-recovery-device  Import recovery device
    overwrite-server-url    Change the server URL for the device, this is normally not needed
    help                    Print this message or the help of the given subcommand(s)

    Options:
    -h, --help  Print help

or subcommand

.. code-block:: shell

>     ./parsec-cli 3.8.2-a.0+dev  device list --help

Reviewer: ```suggestion
    parsec-cli device list --help
```

    List all devices

>     Usage: parsec-cli-3.9.0-rc.1-x86_64-unknown-linux-musl device list [OPTIONS]

Reviewer: ```suggestion
    Usage: parsec-cli device list [OPTIONS]
```

    Options:
    -c, --config-dir <CONFIG_DIR>  Parsec config directory [env: PARSEC_CONFIG_DIR=] [default: /home/aurelia/.config/parsec3/libparsec]
    -h, --help                     Print help

Commonly used parameters
------------------------

- ``--device``, ``-d``: the short id of the device used to authenticate the operation.

> - ``--password-stdin``: to read the password from stdin

Reviewer: I'm not sure this is really helpful here without further explanation.

I would either explain a little bit further why & when these options are needed OR remove the whole section.

> - --password-stdin: to read the password from stdin

Reviewer 2: **todo:** The options need to be highlighted in code span
